src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -387,21 +405,41 @@ export default {
 			}
 		},
 
+		token() {
+			return this.$store.getters.getToken()
+		},
+
+		getFilterCaption() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'Unread mentions')
+				: t('spreed', 'Unread messages')
+		},
+
+		messageDisplay() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'You have no unread mentions in your inbox')
+				: t('spreed', 'You have no unread messages in your inbox')
+		},
+
 		filteredConversationsList() {
 			if (this.isFocused) {
 				return this.conversationsList
 			}
-
-			if (this.isFiltered === 'unread') {
-				return this.conversationsList.filter(conversation => conversation.unreadMessages > 0
-				|| conversation.hasCall || conversation.token === this.$store.getters.getToken())
-			}
-
-			if (this.isFiltered === 'mentions') {
-				return this.conversationsList.filter(conversation => conversation.unreadMention
-				|| conversation.hasCall
-				|| (conversation.unreadMessages > 0 && (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER))
-				|| conversation.token === this.$store.getters.getToken())
+			// applying filters
+			if (this.isFiltered) {
+				const filteredConversations = this.conversationsList.filter(function(conversation) {
+					return this.filterFunction(conversation)
+								|| conversation.hasCall
+								|| conversation.token === this.token
+				}.bind(this))
+
+				// return empty if it only includes the current conversation without any flags
+				const includesCurrentConversationOnly = filteredConversations.length === 1
+					&& filteredConversations.filter(function(conversation) {
+						return !this.filterFunction(conversation) && conversation.token === this.token
+					}.bind(this)).length
+
+				return includesCurrentConversationOnly ? [] : filteredConversations
```

Comment:
Same for the hasCall. Then in filter function we can count amount of `this.filterFunction(conversation) === true`, and if it's equal 0, show EmptyContent. Thus, current conversation and ongoing call will still be always visible

Reply:
It seems super confusing to have the empty content and the current conversation at the same time. Also, if I filter and find the current conversation, I have a thought that there must be a new message in the conversation that came while it is open.

Reply:
I would agree with @DorraJaouad here, since the current conversation is open in the main content anyway, we don't need to show it in the results :) Even when we search for a conversation using the search bar, the results don't include the current conversation, so we can follow that same logic for filters too :)

Reply:
Note this was specifically requested from the design team:
https://github.com/nextcloud/spreed/issues/10498

cc @marcoambrosini

Reply:
Isn't it marked as read only when you scroll fully to the bottom? So when we open a conversation with some unread messages it would still be unread right? In that case it would show up in the list anyway.

Reply:
Depending on amount of unread messages, it may be instantly triggered

Reply:
> Note this was specifically requested from the design team:

This is related to the instant disappearance of conversation when marked read, while navigating in the filtered list.  

So, the current situation is  : 
* If there is at least one unread conversation, the current conversation will show in the list ( regardless of its unread flag) and will be removed when you navigate away. 
* If there are 0 unread conversations, it is an empty list.

Reply:
Suggestion:

- If there are 0 conversations with unread flag, show empty content
- If there is at least one conversation with unread flag, show list of conversations
- Current conversation is shown if there is an unread flag
- When you navigate away from current conversation, if it still has unread flag it stays in the list, if it has no unread flag it is removed from the list 

Does this make sense?

Reply:
To clarify the usecase we have a discussion around:
- I have 3 unread conversations and Lomeapply the filter to show them = 3
- When I read a first conversation, list will be recomputed. Now it includes current conversation + 2 unread ones = 3
- Read next - previous current removed from list - now I see current + 1 unread = 2
- Read last - should see only current = 1
- :warning: but I see empty content instead = 0

Reply:
> Read last - should see only current = 1

yes

> ⚠️ but I see empty content instead = 0

You will see it when you click on the talk icon (top bar my main usecase for this PR here, just to have a reminder that it's still filtered) or are removed from the room.

Reply:
[Screencast from 15.10.2023 13:58:13.webm](https://github.com/nextcloud/spreed/assets/93392545/c4b6b6d2-d3fa-47e9-b60b-0d73d24f406e)

Same haapens when you marked last conversation as read manually, or it's triggered by itself

Reply:
Then, it should broken down as follows:
* At the moment the filter is first applied:

    - Check if the filtered list contains only the currently active conversation.
    - If so, hide the current conversation by returning an empty list.
    - This handles the case of clicking the filter initially and showing empty content if there are 0 unread conversations

* After filters are applied (during subsequent filter navigation):

    - The current active conversation should always be shown (regardless of its flag).
    - This will solve the disappearance of the last-read conversation.

https://github.com/nextcloud/spreed/assets/84044328/97f83aa0-3279-4cf3-889c-15b5f8bdeff9